sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcColumnarBatchReaderSuite.scala

```diff
@@ -77,4 +77,64 @@ class OrcColumnarBatchReaderSuite extends QueryTest with SharedSparkSession {
       assert(p1.getUTF8String(0) === partitionValues.getUTF8String(0))
     }
   }
+
+  test("SPARK-32234: orc data created by the hive tables having _col fields name") {
+    var error: Throwable = null
+    withTable("test_date_hive_orc") {
+      spark.sql(
+        """
+          |CREATE TABLE `test_date_hive_orc`
+          | (`_col1` INT,`_col2` STRING,`_col3` INT)
+          |  USING orc
+        """.stripMargin)
+      spark.sql(
+        """insert into
+          | test_date_hive_orc
+          |  values(9, '12', 2020)
+        """.stripMargin)
+      try {
+        val df = spark.sql("select _col2 from test_date_hive_orc")
+        checkAnswer(df, Row("12"))
+      } catch {
+        case e: Throwable =>
+          error = e
+      }
+      assert(error == null)
+      spark.sql(
+        s"""
+           |DROP TABLE IF
```

Comment:
you don't need to drop the table. Please see the implementation of `withTable`.

Reply:
yes in withTable table its handled, Removed this drop table .